docker/daemon.go

```diff
@@ -81,7 +84,20 @@ func mainDaemon() {
 		return
 	}
 	eng := engine.New()
-	signal.Trap(eng.Shutdown)
+	sig.Trap(eng.Shutdown)
+	if *flDebug {
+		go func() {
+			sigs := make(chan os.Signal, 1)
+			signal.Notify(sigs, syscall.SIGQUIT)
+			buf := make([]byte, 16384)
+			for {
+				<-sigs
+				buf = buf[:runtime.Stack(buf, true)]
+				log.Debugf("=== BEGIN Docker daemon goroutine stacktrace ===\n\n")
+				log.Debugf("%s\n=== END Docker daemon goroutine stacktrace ===\n", buf)
```

Comment:
Probably makes sense to write one big log message. Because actually other messages can be between :)